source/common/router/retry_state_impl.cc

```diff
@@ -290,9 +293,9 @@ RetryStatus RetryStateImpl::shouldHedgeRetryPerTryTimeout(DoRetryCallback callba
 }
 
 bool RetryStateImpl::wouldRetryFromHeaders(const Http::ResponseHeaderMap& response_headers) {
-  // We never retry if the request is rate limited.
+  // We retry our own rate limited requests only when the envoy-ratelimited policy is in effect.
```

Comment:
"Our own" implies to me that this envoy instance rate limited the request, but its likely going to be another Envoy instance

Reply:
Yes, that's a good point. I'll rephrase this to make it clear we're talking about an upstream envoy.